Tue Dec 27 22:08:50 FET 2011  Oleg Osipovich <swarm.dev@gmail.com>
  * Added type signatures for dumpPatch and dumpTag.
diff -rN -u old-darcs-fastconvert/Export.hs new-darcs-fastconvert/Export.hs
--- old-darcs-fastconvert/Export.hs	2011-12-27 22:11:21.534669730 +0300
+++ new-darcs-fastconvert/Export.hs	2011-12-27 22:11:21.535669730 +0300
@@ -80,6 +80,8 @@
                   dumpFiles subs
   when (not isfile && not isdir) $ liftIO $ putStrLn $ "D " ++ anchorPath "" file

+dumpPatch :: (RepoPatch p) => (PatchInfoAnd p -> Int -> TreeIO ()) -> PatchInfoAnd p
+             -> Int -> TreeIO ()
 dumpPatch mark p n =
   do dumpBits [ BLC.pack $ "progress " ++ show n ++ ": " ++ patchName p
               , "commit refs/heads/master" ]
@@ -89,6 +91,7 @@
               , patchMessage p ]
      when (n > 1) $ dumpBits [ BLU.fromString $ "from :" ++ show (n - 1) ]

+dumpTag :: (RepoPatch p) => PatchInfoAnd p -> Int -> TreeIO ()
 dumpTag p n =
   dumpBits [ BLU.fromString $ "progress TAG " ++ tagName p
            , BLU.fromString $ "tag " ++ tagName p -- FIXME is this valid?
